### PR TITLE
Change password endpoint

### PR DESCRIPTION
**Pull request by lalagvana**

### CODE REVIEW

File: src/UserService.Data/UserCredentialsRepository.cs [hunk 1]
```diff
@@ -65,5 +67,21 @@ public void CreateUserCredentials(DbUserCredentials userCredentials)
...
             provider.Save();
         }
+
+        public void ChangePassword(string login, string newPassword)
+        {
+            DbUserCredentials userCredentials = provider.UserCredentials.FirstOrDefault(uc => uc.Login == login);
+
+            if (userCredentials == null)
+            {
+                throw new Exception("User credentials with this user login not found.");
```
Comment (DarkFelas) [inline src/UserService.Data/UserCredentialsRepository.cs:77]: Kernel NotFoundException

File: src/UserService.Data/UserCredentialsRepository.cs [hunk 2]
```diff
@@ -65,5 +67,21 @@ public void CreateUserCredentials(DbUserCredentials userCredentials)
...
+            }
+
+            userCredentials.PasswordHash = Encoding.UTF8.GetString(new SHA512Managed().ComputeHash(
```
Comment (DarkFelas) [inline src/UserService.Data/UserCredentialsRepository.cs:80]: What about new hash generation logic?

File: test/UserService.Data.UnitTests/UserCredentialsRepositoryTests.cs [hunk 3]
```diff
@@ -142,5 +149,24 @@ public void ShouldCreatedUserCredentialsSuccessful()
             SerializerAssert.AreEqual(newUserCredentials,
                 provider.UserCredentials.FirstOrDefault(uc => uc.Id == newUserCredentials.Id));
         }
+        #endregion
+
+        #region ChangePassword
+        [Test]
+        public void ShouldThrowExceptionWhenUserCredentialsDoesNotFound()
+        {
+            Assert.Throws<Exception>(() => repository.ChangePassword("login12345", "newPassword"));
```
Comment (DarkFelas) [inline test/UserService.Data.UnitTests/UserCredentialsRepositoryTests.cs:158]: NotFoundException

File: test/UserService.Business.UnitTests/ChangePasswordCommandTests.cs [hunk 4]
```diff
@@ -0,0 +1,10 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace LT.DigitalOffice.UserService.Business.UnitTests
+{
+    class ChangePasswordCommandTests
+    {
```
Comment (DarkFelas) [inline test/UserService.Business.UnitTests/ChangePasswordCommandTests.cs:8]: Tests???

**Review by DarkFelas — CHANGES_REQUESTED** [review]

**Review by KostinP — APPROVED** [review]

File: src/UserService.Business/ChangePasswordCommand.cs [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+        public ChangePasswordCommand(
+            [FromServices] IUserCredentialsRepository repository)
+        {
+            this.repository = repository;
+        }
+
+        public void Execute(ChangePasswordRequest request)
+        {
+            repository.ChangePassword(request.Login, request.NewPassword);
```
Comment (djino97) [inline src/UserService.Business/ChangePasswordCommand.cs:20]: And probably here must need check generatedId from cache(created in ForgotUserPassword)

**Review by DarkFelas — APPROVED** [review]

**Pull request merged**